### PR TITLE
Fixing version checker and E2E

### PR DESCRIPTION
**Pull request by chrislovecnm**
The version checker was implemented to only support versions
that are defined in the RELATED_IMAGES, but when using
image.name a user needs to have the capability to run any
version of CRDB.

Fixed e2e and ovehauled e2e

- updated kind
- removed tests that were duplicates
- tests that can run in parrallel are now doing it
- updated tested code to allow for new changes with version
checking

### CODE REVIEW

**Comment by chrislovecnm** [comment]
Getting this error at times.  @alinadonisa 

```
"Crdb Version Validator", "CrdbCluster": "crdb-test-bm92k5/crdb", "error": "pod is not running for job: crdb-version-checker: job pods are not running yet waiting longer", "errorVerbose": "job pods are not running yet waiting longer\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.LogError\n\tpkg/actor/validate_version.go:331\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.IsJobPodRunning\n\tpkg/actor/validate_version.go:302\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.WaitUntilJobPodIsRunning.func1\n\tpkg/actor/validate_version.go:317\ngithub.com/cenkalti/backoff.RetryNotify\n\texternal/com_github_cenkalti_backoff/retry.go:37\ngithub.com/cenkalti/backoff.Retry\n\texternal/com_github_cenkalti_backoff/retry.go:24\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.WaitUntilJobPodIsRunning\n\tpkg/actor/validate_version.go:322\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.(*versionChecker).Act\n\tpkg/actor/validate_version.go:154\ngithub.com/cockroachdb/cockroach-operator/pkg/controller.(*ClusterReconciler).Reconcile\n\tpkg/controller/cluster_controller.go:126\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\texternal/io_k8s_sigs_controller_runtime/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\texternal/io_k8s_sigs_controller_runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\texternal/io_k8s_sigs_controller_runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:90\nruntime.goexit\n\tGOROOT/src/runtime/asm_amd64.s:1373\npod is not running for job: crdb-version-checker\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.WaitUntilJobPodIsRunning\n\tpkg/actor/validate_version.go:323\ngithub.com/cockroachdb/cockroach-operator/pkg/actor.(*versionChecker).Act\n\tpkg/actor/validate_version.go:154\ngithub.com/cockroachdb/cockroach-operator/pkg/controller.(*ClusterReconciler).Reconcile\n\tpkg/controller/cluster_controller.go:126\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\texternal/io_k8s_sigs_controller_runtime/pkg/internal/controller/controller.go:256\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\texternal/io_k8s_sigs_controller_runtime/pkg/internal/controller/controller.go:232\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\texternal/io_k8s_sigs_controller_runtime/pkg/internal/controller/controller.go:211\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:155\nk8s.io/apimachinery/pkg/util/wait.BackoffUntil\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:156\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:133\nk8s.io/apimachinery/pkg/util/wait.Until\n\texternal/io_k8s_apimachinery/pkg/util/wait/wait.go:90\nruntime.goexit\n\tGOROOT/src/runtime/asm_amd64.s:1373"}
```

File: e2e/e2e_test.go [hunk 1]
```diff
@@ -146,21 +122,28 @@ func TestCreatesSecureClusterWithGeneratedCert(t *testing.T) {
...
 	sb.StartManager(t, controller.InitClusterReconcilerWithLogger(testLog))
 
-	b := testutil.NewBuilder("crdb").WithNodeCount(1).WithTLS().WithEmptyDirDataStore()
+	builder := testutil.NewBuilder("crdb").WithNodeCount(3).WithTLS().
+		WithImage("cockroachdb/cockroach:v20.2.5").
+		WithPVDataStore("1Gi", "standard" /* default storage class in KIND */)
 
 	create := Step{
-		name: "creates 1-node secure cluster",
+		name: "creates 3-node insecure cluster",
```
Comment (keith-mcclellan) [inline e2e/e2e_test.go:130]: This function is "TestCreatesSecureClusterWithGeneratedCert" but its creating an insecure cluster?

File: e2e/e2e_test.go [hunk 2]
```diff
@@ -288,61 +278,9 @@ func TestUpgradesMajorVersion19to20(t *testing.T) {
 }
 
 func TestUpgradesMajorVersion19_1To19_2(t *testing.T) {
```
Comment (keith-mcclellan) [inline e2e/e2e_test.go:280]: We should not have a hardcoded version upgrade test, this needs to be changed. In the near term, lets move this to testing upgrading from 19.2 to 20.1 but in the future this should be dynamic

Comment (keith-mcclellan) [reply e2e/e2e_test.go:280]: 19.1 is no longer supported anyway

File: e2e/assert.go [hunk 3]
```diff
@@ -33,39 +35,68 @@ import (
...
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apiresource "k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-func requireClusterToBeReadyEventually(t *testing.T, sb testenv.DiffingSandbox, b testutil.ClusterBuilder) {
+func RequireClusterToBeReadyEventuallyTimeout(t *testing.T, sb testenv.DiffingSandbox, b testutil.ClusterBuilder, timeout time.Duration) {
```
Comment (keith-mcclellan) [inline e2e/assert.go:48]: Please add a comment here

File: e2e/e2e_test.go [hunk 4]
```diff
@@ -99,42 +106,11 @@ func TestMain(m *testing.M) {
...
-			requireClusterToBeReadyEventually(t, sb, b)
-
-			state, err := sb.Diff()
-			require.NoError(t, err)
-
-			expected := testutil.ReadOrUpdateGoldenFile(t, state, *updateOpt)
-
-			testutil.AssertDiff(t, expected, state)
-		},
+func TestCreatesSecureCluster(t *testing.T) {
```
Comment (keith-mcclellan) [inline e2e/e2e_test.go:110]: this is pretty self explanatory but please add a comment for consistency sake

**Review by keith-mcclellan — CHANGES_REQUESTED** [review]
Functionally, some of these tests (19.1 <> 19.2) are no longer valid.  In the future we need to make them dynamic but for this PR lets at least move them to 19.2<>20.1 and 20.1<>20.2

File: e2e/e2e_test.go (hunk 4 above)
Comment (chrislovecnm) [reply e2e/e2e_test.go:110]: You don't usually add comments on test funcs in go ;). It is outside of the go standard.  I will add some comments inside the func.

File: e2e/e2e_test.go (hunk 1 above)
Comment (chrislovecnm) [reply e2e/e2e_test.go:130]: typo

**Review by keith-mcclellan — APPROVED** [review]
LGTM

**Pull request merged**